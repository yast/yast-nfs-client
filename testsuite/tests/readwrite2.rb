# encoding: utf-8

# Module:
#   NFS client configuration
#
# Summary:
#   Testsuite
#
# Authors:
#   Martin Vidner <mvidner@suse.cz>
#
# $Id$
module Yast
  class Readwrite2Client < Client
    def main
      # testedfiles: Nfs.ycp Service.ycp Report.ycp Testsuite.ycp

      Yast.include self, "testsuite.rb"

      @I_READ = { "target" => { "size" => 0 } }
      @I_WRITE = {}
      @I_EXEC = { "target" => { "bash_output" => {} } }
      TESTSUITE_INIT([@I_READ, @I_WRITE, @I_EXEC], nil)

      Yast.import "Nfs"
      Yast.import "Report"
      Yast.import "Progress"

      @progress_orig = Progress.set(false)
      Report.DisplayErrors(false, 0)

      @service_on = { "start" => ["3", "5"], "stop" => ["3", "5"] }
      @service_off = { "start" => [], "stop" => [] }
      @READ = {
        # Runlevel:
        "init"      => {
          "scripts" => {
            "exists"   => true,
            "runlevel" => { "portmap" => @service_on, "nfs" => @service_on },
            # their contents is not important for ServiceAdjust
            "comment"  => {
              "portmap" => {},
              "nfs"     => {}
            }
          }
        },
        # Nis itself:
        "etc"       => {
          "fstab"       => [
            {
              "file"    => "/",
              "freq"    => 1,
              "mntops"  => "defaults",
              "passno"  => 2,
              "spec"    => "/dev/hda6",
              "vfstype" => "reiserfs"
            },
            {
              "file"    => "/home",
              "freq"    => 0,
              "mntops"  => "defaults",
              "passno"  => 0,
              "spec"    => "foo.bar.com:/home",
              "vfstype" => "nfs"
            },
            {
              "file"    => "/var/spool/mail",
              "freq"    => 0,
              "mntops"  => "defaults",
              "passno"  => 0,
              "spec"    => "foo.bar.com:/var/spool/mail",
              "vfstype" => "nfs"
            },
            {
              "file"    => "/a\\040space",
              "freq"    => 1,
              "mntops"  => "defaults",
              "passno"  => 3,
              "spec"    => "/dev/hda7",
              "vfstype" => "reiserfs"
            },
            {
              "file"    => "/b\\040space",
              "freq"    => 0,
              "mntops"  => "defaults",
              "passno"  => 0,
              "spec"    => "foo.bar.com:/space\\040dir",
              "vfstype" => "nfs"
            }
          ],
          "idmapd_conf" => "localhost"
        },
        "sysconfig" => {
          "nfs" => { "NFS4_SUPPORT" => "no", "NFS_SECURITY_GSS" => "no" }
        },
        "target"    => { "stat" => { "dummy" => true } }
      }

      # services portmap & nfs are stopped.
      @READ3 = Builtins.eval(@READ)
      Ops.set(@READ3, ["init", "scripts", "runlevel", "portmap"], @service_off)
      Ops.set(@READ3, ["init", "scripts", "runlevel", "nfs"], @service_off)

      # no nfs file-systems in /etc/fstab
      @EMPTY = Builtins.eval(@READ)
      Ops.set(
        @EMPTY,
        ["etc", "fstab"],
        [
          {
            "file"    => "/",
            "freq"    => 1,
            "mntops"  => "defaults",
            "passno"  => 2,
            "spec"    => "/dev/hda6",
            "vfstype" => "ext3"
          }
        ]
      )

      # services portmap & nfs are stopped; and /etc/fstab contains no nfs imports
      @EMPTY3 = Builtins.eval(@READ3)
      Ops.set(
        @EMPTY3,
        ["etc", "fstab"],
        [
          {
            "file"    => "/",
            "freq"    => 1,
            "mntops"  => "defaults",
            "passno"  => 2,
            "spec"    => "/dev/hda6",
            "vfstype" => "ext3"
          }
        ]
      )

      @WRITE = {}

      @WRITE_KO = { "etc" => { "fstab" => false } }

      @EXECUTE = {
        "target" => {
          "bash_output" => { "exit" => 0, "stdout" => "", "stderr" => "" },
          "mkdir"       => true
        }
      }

      # fstab contains nfs mounts & services are running
      DUMP("\nRead  - nfs is in use & running\n")
      TEST(->() { Nfs.Read }, [@READ, @WRITE, @EXECUTE], nil)
      # fstab unchanged
      DUMP("\nWrite - nfs is in use - start services\n")
      TEST(->() { Nfs.Write }, [@READ, @WRITE, @EXECUTE], nil)

      # fstab contains nfs mounts & services are stopped:-(
      DUMP("\nRead  - nfs is in use & stopped\n")
      TEST(->() { Nfs.Read }, [@READ3, @WRITE, @EXECUTE], nil)
      # fstab unchanged - so, start services
      DUMP("\nWrite - nfs is in use - so, start services\n")
      TEST(->() { Nfs.Write }, [@READ3, @WRITE, @EXECUTE], nil)

      # fstab contains no nfs mounts, services are running
      DUMP("\nRead  - nfs not used & running\n")
      TEST(->() { Nfs.Read }, [@EMPTY, @WRITE, @EXECUTE], nil)
      # fstab unchanged - so, STOP services
      DUMP("\nWrite - nfs not used - so, stopping services\n")
      TEST(->() { Nfs.Write }, [@EMPTY, @WRITE, @EXECUTE], nil)

      # fstab contains no nfs mount, serives are stopped
      DUMP("\nRead  - nfs not used & services are stopped\n")
      TEST(->() { Nfs.Read }, [@EMPTY3, @WRITE, @EXECUTE], nil)
      # fstab unchanged - so, leave services stopped
      DUMP("\nWrite - nfs not used; leave services stopped\n")
      TEST(->() { Nfs.Write }, [@EMPTY3, @WRITE, @EXECUTE], nil) 

      # // nfs and portmap are running
      #     DUMP ("\nRead  - services are running\n");
      #     TEST (``(Nfs::Read ()), [READ, WRITE, EXECUTE], nil);
      #     DUMP ("\nWrite - services will be stopped\n");
      #     // Stop services!
      # //    Nfs::start = false;
      #     // And Write
      #     TEST (``(Nfs::Write ()), [READ, WRITE, EXECUTE], nil);
      #
      #     // nfs and portmap are running
      #     DUMP ("\nRead  - services are running\n");
      #     TEST (``(Nfs::Read ()), [READ, WRITE, EXECUTE], nil);
      #     DUMP ("\nWrite - services are running\n");
      #     // Start services (nfsserver)
      # //    Nfs::start = true;
      #     // And Write
      #     TEST (``(Nfs::Write ()), [READ, WRITE, EXECUTE], nil);
      #
      #     // nfs and portmap are stopped
      #     DUMP ("\nRead  - services are stopped\n");
      #     TEST (``(Nfs::Read ()), [READ3, WRITE, EXECUTE], nil);
      #     DUMP ("\nWrite - services will be stopped\n");
      #     // Leave services stopped
      # //    Nfs::start = false;
      #     // And Write
      #     TEST (``(Nfs::Write ()), [READ3, WRITE, EXECUTE], nil);
      #
      #     // nfs and portmap are stopped
      #     DUMP ("\nRead  - services are stopped\n");
      #     TEST (``(Nfs::Read ()), [READ3, WRITE, EXECUTE], nil);
      #     DUMP ("\nWrite - services will be started\n");
      #     // Start services
      # //    Nfs::start = true;
      #     // And Write
      #     TEST (``(Nfs::Write ()), [READ3, WRITE, EXECUTE], nil);
      #
      #     DUMP ("\nEMPTY\n");
      #     TEST (``(Nfs::Read ()), [EMPTY, WRITE, EXECUTE], nil);
      #     TEST (``(Nfs::Write ()), [EMPTY, WRITE, EXECUTE], nil);

      nil
    end
  end
end

Yast::Readwrite2Client.new.main
