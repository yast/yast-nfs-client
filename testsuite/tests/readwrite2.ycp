/**
 * Module:
 *   NFS client configuration
 *
 * Summary:
 *   Testsuite
 *
 * Authors:
 *   Martin Vidner <mvidner@suse.cz>
 *
 * $Id$
 */
{
    // testedfiles: Nfs.ycp Service.ycp Report.ycp Testsuite.ycp

    include "testsuite.ycp";

    map I_READ = $[
	"target" : $[
	    "size" : 0,
	],
    ];
    map I_WRITE = $[];
    map I_EXEC = $[
	"target" : $[
	    "bash_output" : $[],
	],
    ];
    TESTSUITE_INIT ([I_READ, I_WRITE, I_EXEC], nil);

    import "Nfs";
    import "Report";
    import "Progress";

    boolean progress_orig = Progress::set (false);
    Report::DisplayErrors (false, 0);

    map service_on = $[
	"start": [ "3", "5"],
	"stop":	 [ "3", "5"],
	];
    map service_off = $[
	"start": [],
	"stop":	 [],
	];
    map READ = $[
	// Runlevel:
	"init": $[
	    "scripts": $[
		"exists": true,
		"runlevel": $[
		    "portmap": service_on,
		    "nfs": service_on,
		    ],
		// their contents is not important for ServiceAdjust
		"comment": $[
		    "portmap": $[],
		    "nfs": $[],
		    ],
		],
	    ],
    /*
	// targetpkg:
	"targetpkg": $[
	    // autofs
	    "installed": true,
	    ],
    */
	// Nis itself:
	"etc": $[
	    "fstab": [
		$[
		    "file":"/",
		    "freq":1,
		    "mntops":"defaults",
		    "passno":2,
		    "spec":"/dev/hda6",
		    "vfstype":"reiserfs",
		],
		$[
		    "file":"/home",
		    "freq":0,
		    "mntops":"defaults",
		    "passno":0,
		    "spec":"foo.bar.com:/home",
		    "vfstype":"nfs",
		],
		$[
		    "file":"/var/spool/mail",
		    "freq":0,
		    "mntops":"defaults",
		    "passno":0,
		    "spec":"foo.bar.com:/var/spool/mail",
		    "vfstype":"nfs",
		],
		$[
		    "file":"/a\\040space",
		    "freq":1,
		    "mntops":"defaults",
		    "passno":3,
		    "spec":"/dev/hda7",
		    "vfstype":"reiserfs",
		],
		$[
		    "file":"/b\\040space",
		    "freq":0,
		    "mntops":"defaults",
		    "passno":0,
		    "spec":"foo.bar.com:/space\\040dir",
		    "vfstype":"nfs",
		],
	      ],
	      "idmapd_conf" : "localhost"
	    ],
	"sysconfig": $[
	     "nfs" : $[
	         "NFS4_SUPPORT": "no",
                 "NFS_SECURITY_GSS": "no",
	     ],
	 ],
	];

    // services portmap & nfs are stopped.
    map READ3 = (map) eval (READ);
    READ3["init","scripts","runlevel","portmap"] = service_off;
    READ3["init","scripts","runlevel","nfs"]     = service_off;

    // no nfs file-systems in /etc/fstab
    map EMPTY = (map) eval (READ);
    EMPTY["etc","fstab"] = [
	$[
	    "file":"/",
	    "freq":1,
	    "mntops":"defaults",
	    "passno":2,
	    "spec":"/dev/hda6",
	    "vfstype":"ext3",
	],
    ];

    // services portmap & nfs are stopped; and /etc/fstab contains no nfs imports
    map EMPTY3 = (map) eval (READ3);
    EMPTY3["etc","fstab"] = [
	$[
	    "file":"/",
	    "freq":1,
	    "mntops":"defaults",
	    "passno":2,
	    "spec":"/dev/hda6",
	    "vfstype":"ext3",
	],
    ];

    map WRITE = $[
	];

    map WRITE_KO = $[
	"etc": $[
	    "fstab": false,
	    ],
	];

    map EXECUTE = $[
	"target": $[
	    "bash_output": $[
		"exit": 0,
		"stdout": "",
		"stderr": "",
		],
	    "mkdir": true,
	    ],
	];

    // fstab contains nfs mounts & services are running
    DUMP ("\nRead  - nfs is in use & running\n");
    TEST (``(Nfs::Read ()), [READ, WRITE, EXECUTE], nil);
    // fstab unchanged
    DUMP ("\nWrite - nfs is in use - start services\n");
    TEST (``(Nfs::Write ()), [READ, WRITE, EXECUTE], nil);

    // fstab contains nfs mounts & services are stopped:-(
    DUMP ("\nRead  - nfs is in use & stopped\n");
    TEST (``(Nfs::Read ()), [READ3, WRITE, EXECUTE], nil);
    // fstab unchanged - so, start services
    DUMP ("\nWrite - nfs is in use - so, sart services\n");
    TEST (``(Nfs::Write ()), [READ3, WRITE, EXECUTE], nil);

    // fstab contains no nfs mounts, services are running
    DUMP ("\nRead  - nfs not used & running\n");
    TEST (``(Nfs::Read ()), [EMPTY, WRITE, EXECUTE], nil);
    // fstab unchanged - so, STOP services
    DUMP ("\nWrite - nfs not used - so, stopping services\n");
    TEST (``(Nfs::Write ()), [EMPTY, WRITE, EXECUTE], nil);

    // fstab contains no nfs mount, serives are stopped
    DUMP ("\nRead  - nfs not used & services are stopped\n");
    TEST (``(Nfs::Read ()), [EMPTY3, WRITE, EXECUTE], nil);
    // fstab unchanged - so, leave services stopped
    DUMP ("\nWrite - nfs not used; leave services stopped\n");
    TEST (``(Nfs::Write ()), [EMPTY3, WRITE, EXECUTE], nil);

/*    // nfs and portmap are running
    DUMP ("\nRead  - services are running\n");
    TEST (``(Nfs::Read ()), [READ, WRITE, EXECUTE], nil);
    DUMP ("\nWrite - services will be stopped\n");
    // Stop services!
//    Nfs::start = false;
    // And Write
    TEST (``(Nfs::Write ()), [READ, WRITE, EXECUTE], nil);

    // nfs and portmap are running
    DUMP ("\nRead  - services are running\n");
    TEST (``(Nfs::Read ()), [READ, WRITE, EXECUTE], nil);
    DUMP ("\nWrite - services are running\n");
    // Start services (nfsserver)
//    Nfs::start = true;
    // And Write
    TEST (``(Nfs::Write ()), [READ, WRITE, EXECUTE], nil);

    // nfs and portmap are stopped
    DUMP ("\nRead  - services are stopped\n");
    TEST (``(Nfs::Read ()), [READ3, WRITE, EXECUTE], nil);
    DUMP ("\nWrite - services will be stopped\n");
    // Leave services stopped
//    Nfs::start = false;
    // And Write
    TEST (``(Nfs::Write ()), [READ3, WRITE, EXECUTE], nil);

    // nfs and portmap are stopped
    DUMP ("\nRead  - services are stopped\n");
    TEST (``(Nfs::Read ()), [READ3, WRITE, EXECUTE], nil);
    DUMP ("\nWrite - services will be started\n");
    // Start services
//    Nfs::start = true;
    // And Write
    TEST (``(Nfs::Write ()), [READ3, WRITE, EXECUTE], nil);

    DUMP ("\nEMPTY\n");
    TEST (``(Nfs::Read ()), [EMPTY, WRITE, EXECUTE], nil);
    TEST (``(Nfs::Write ()), [EMPTY, WRITE, EXECUTE], nil);*/
}
