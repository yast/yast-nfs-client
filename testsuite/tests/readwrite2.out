Dump	
Dump	Read  - nfs is in use & running
Dump	
Read	.etc.fstab [$["file":"/", "freq":1, "mntops":"defaults", "passno":2, "spec":"/dev/hda6", "vfstype":"reiserfs"], $["file":"/home", "freq":0, "mntops":"defaults", "passno":0, "spec":"foo.bar.com:/home", "vfstype":"nfs"], $["file":"/var/spool/mail", "freq":0, "mntops":"defaults", "passno":0, "spec":"foo.bar.com:/var/spool/mail", "vfstype":"nfs"], $["file":"/a\\040space", "freq":1, "mntops":"defaults", "passno":3, "spec":"/dev/hda7", "vfstype":"reiserfs"], $["file":"/b\\040space", "freq":0, "mntops":"defaults", "passno":0, "spec":"foo.bar.com:/space\\040dir", "vfstype":"nfs"]]
Read	.sysconfig.nfs.NFS4_SUPPORT "no"
Read	.sysconfig.nfs.NFS_SECURITY_GSS "no"
Read	.etc.idmapd_conf "Domain" "localhost"
Read	.init.scripts.exists "SuSEfirewall2_init" true
Read	.init.scripts.runlevel "SuSEfirewall2_init" $["nfs":$["start":["3", "5"], "stop":["3", "5"]], "portmap":$["start":["3", "5"], "stop":["3", "5"]]]
Read	.init.scripts.exists "SuSEfirewall2_init" true
Execute	.target.bash "/etc/init.d/SuSEfirewall2_init status" $["TERM":"raw"] 0
Return	true
Dump	
Dump	Write - nfs is in use - start services
Dump	
Read	.etc.fstab [$["file":"/", "freq":1, "mntops":"defaults", "passno":2, "spec":"/dev/hda6", "vfstype":"reiserfs"], $["file":"/home", "freq":0, "mntops":"defaults", "passno":0, "spec":"foo.bar.com:/home", "vfstype":"nfs"], $["file":"/var/spool/mail", "freq":0, "mntops":"defaults", "passno":0, "spec":"foo.bar.com:/var/spool/mail", "vfstype":"nfs"], $["file":"/a\\040space", "freq":1, "mntops":"defaults", "passno":3, "spec":"/dev/hda7", "vfstype":"reiserfs"], $["file":"/b\\040space", "freq":0, "mntops":"defaults", "passno":0, "spec":"foo.bar.com:/space\\040dir", "vfstype":"nfs"]]
Execute	.target.mkdir "/home" true
Execute	.target.mkdir "/var/spool/mail" true
Execute	.target.mkdir "/b space" true
Execute	.target.bash "/bin/cp $ORIG $BACKUP" $["BACKUP":"/etc/fstab.YaST2.save", "ORIG":"/etc/fstab"] 0
Write	.etc.fstab [$["file":"/", "freq":1, "mntops":"defaults", "passno":2, "spec":"/dev/hda6", "vfstype":"reiserfs"], $["file":"/a\\040space", "freq":1, "mntops":"defaults", "passno":3, "spec":"/dev/hda7", "vfstype":"reiserfs"], $["file":"/home", "freq":0, "mntops":"defaults", "passno":0, "spec":"foo.bar.com:/home", "vfstype":"nfs"], $["file":"/var/spool/mail", "freq":0, "mntops":"defaults", "passno":0, "spec":"foo.bar.com:/var/spool/mail", "vfstype":"nfs"], $["file":"/b\\040space", "freq":0, "mntops":"defaults", "passno":0, "spec":"foo.bar.com:/space\\040dir", "vfstype":"nfs"]] true
Read	.init.scripts.exists "portmap" true
Read	.init.scripts.exists "portmap" true
Read	.init.scripts.exists "portmap" true
Read	.init.scripts.runlevel "portmap" $["nfs":$["start":["3", "5"], "stop":["3", "5"]], "portmap":$["start":["3", "5"], "stop":["3", "5"]]]
Read	.init.scripts.comment "portmap" $["nfs":$[], "portmap":$[]]
Read	.init.scripts.exists "nfs" true
Read	.init.scripts.exists "nfs" true
Read	.init.scripts.exists "nfs" true
Read	.init.scripts.runlevel "nfs" $["nfs":$["start":["3", "5"], "stop":["3", "5"]], "portmap":$["start":["3", "5"], "stop":["3", "5"]]]
Read	.init.scripts.comment "nfs" $["nfs":$[], "portmap":$[]]
Read	.init.scripts.exists "nfsboot" true
Read	.init.scripts.exists "nfsboot" true
Read	.init.scripts.exists "nfsboot" true
Read	.init.scripts.runlevel "nfsboot" $["nfs":$["start":["3", "5"], "stop":["3", "5"]], "portmap":$["start":["3", "5"], "stop":["3", "5"]]]
Read	.init.scripts.comment "nfsboot" $["nfs":$[], "portmap":$[]]
Execute	.target.bash_output "/sbin/insserv -d /etc/init.d/nfsboot" $["exit":0, "stderr":"", "stdout":""]
Write	.sysconfig.nfs.NFS4_SUPPORT "no" true
Write	.sysconfig.nfs.NFS_SECURITY_GSS "no" true
Read	.init.scripts.exists "nfs" true
Execute	.target.bash_output "/etc/init.d/nfs stop" $["TERM":"raw"] $["exit":0, "stderr":"", "stdout":""]
Read	.init.scripts.exists "portmap" true
Execute	.target.bash "/etc/init.d/portmap status" $["TERM":"raw"] 0
Read	.init.scripts.exists "nfs" true
Execute	.target.bash_output "/etc/init.d/nfs start" $["TERM":"raw"] $["exit":0, "stderr":"", "stdout":""]
Read	.init.scripts.exists "nfs" true
Execute	.target.bash "/etc/init.d/nfs status" $["TERM":"raw"] 0
Read	.init.scripts.exists "SuSEfirewall2_init" true
Execute	.target.bash "/etc/init.d/SuSEfirewall2_init status" $["TERM":"raw"] 0
Return	true
Dump	
Dump	Read  - nfs is in use & stopped
Dump	
Read	.etc.fstab [$["file":"/", "freq":1, "mntops":"defaults", "passno":2, "spec":"/dev/hda6", "vfstype":"reiserfs"], $["file":"/home", "freq":0, "mntops":"defaults", "passno":0, "spec":"foo.bar.com:/home", "vfstype":"nfs"], $["file":"/var/spool/mail", "freq":0, "mntops":"defaults", "passno":0, "spec":"foo.bar.com:/var/spool/mail", "vfstype":"nfs"], $["file":"/a\\040space", "freq":1, "mntops":"defaults", "passno":3, "spec":"/dev/hda7", "vfstype":"reiserfs"], $["file":"/b\\040space", "freq":0, "mntops":"defaults", "passno":0, "spec":"foo.bar.com:/space\\040dir", "vfstype":"nfs"]]
Read	.sysconfig.nfs.NFS4_SUPPORT "no"
Read	.sysconfig.nfs.NFS_SECURITY_GSS "no"
Read	.etc.idmapd_conf "Domain" "localhost"
Return	true
Dump	
Dump	Write - nfs is in use - so, sart services
Dump	
Read	.etc.fstab [$["file":"/", "freq":1, "mntops":"defaults", "passno":2, "spec":"/dev/hda6", "vfstype":"reiserfs"], $["file":"/home", "freq":0, "mntops":"defaults", "passno":0, "spec":"foo.bar.com:/home", "vfstype":"nfs"], $["file":"/var/spool/mail", "freq":0, "mntops":"defaults", "passno":0, "spec":"foo.bar.com:/var/spool/mail", "vfstype":"nfs"], $["file":"/a\\040space", "freq":1, "mntops":"defaults", "passno":3, "spec":"/dev/hda7", "vfstype":"reiserfs"], $["file":"/b\\040space", "freq":0, "mntops":"defaults", "passno":0, "spec":"foo.bar.com:/space\\040dir", "vfstype":"nfs"]]
Execute	.target.mkdir "/home" true
Execute	.target.mkdir "/var/spool/mail" true
Execute	.target.mkdir "/b space" true
Execute	.target.bash "/bin/cp $ORIG $BACKUP" $["BACKUP":"/etc/fstab.YaST2.save", "ORIG":"/etc/fstab"] 0
Write	.etc.fstab [$["file":"/", "freq":1, "mntops":"defaults", "passno":2, "spec":"/dev/hda6", "vfstype":"reiserfs"], $["file":"/a\\040space", "freq":1, "mntops":"defaults", "passno":3, "spec":"/dev/hda7", "vfstype":"reiserfs"], $["file":"/home", "freq":0, "mntops":"defaults", "passno":0, "spec":"foo.bar.com:/home", "vfstype":"nfs"], $["file":"/var/spool/mail", "freq":0, "mntops":"defaults", "passno":0, "spec":"foo.bar.com:/var/spool/mail", "vfstype":"nfs"], $["file":"/b\\040space", "freq":0, "mntops":"defaults", "passno":0, "spec":"foo.bar.com:/space\\040dir", "vfstype":"nfs"]] true
Read	.init.scripts.exists "portmap" true
Read	.init.scripts.exists "portmap" true
Read	.init.scripts.exists "portmap" true
Read	.init.scripts.runlevel "portmap" $["nfs":$["start":[], "stop":[]], "portmap":$["start":[], "stop":[]]]
Read	.init.scripts.comment "portmap" $["nfs":$[], "portmap":$[]]
Execute	.target.bash_output "/sbin/insserv -d /etc/init.d/portmap" $["exit":0, "stderr":"", "stdout":""]
Read	.init.scripts.exists "nfs" true
Read	.init.scripts.exists "nfs" true
Read	.init.scripts.exists "nfs" true
Read	.init.scripts.runlevel "nfs" $["nfs":$["start":[], "stop":[]], "portmap":$["start":[], "stop":[]]]
Read	.init.scripts.comment "nfs" $["nfs":$[], "portmap":$[]]
Execute	.target.bash_output "/sbin/insserv -d /etc/init.d/nfs" $["exit":0, "stderr":"", "stdout":""]
Read	.init.scripts.exists "nfsboot" true
Read	.init.scripts.exists "nfsboot" true
Read	.init.scripts.exists "nfsboot" true
Read	.init.scripts.runlevel "nfsboot" $["nfs":$["start":[], "stop":[]], "portmap":$["start":[], "stop":[]]]
Read	.init.scripts.comment "nfsboot" $["nfs":$[], "portmap":$[]]
Execute	.target.bash_output "/sbin/insserv -d /etc/init.d/nfsboot" $["exit":0, "stderr":"", "stdout":""]
Write	.sysconfig.nfs.NFS4_SUPPORT "no" true
Write	.sysconfig.nfs.NFS_SECURITY_GSS "no" true
Read	.init.scripts.exists "nfs" true
Execute	.target.bash_output "/etc/init.d/nfs stop" $["TERM":"raw"] $["exit":0, "stderr":"", "stdout":""]
Read	.init.scripts.exists "portmap" true
Execute	.target.bash "/etc/init.d/portmap status" $["TERM":"raw"] 0
Read	.init.scripts.exists "nfs" true
Execute	.target.bash_output "/etc/init.d/nfs start" $["TERM":"raw"] $["exit":0, "stderr":"", "stdout":""]
Read	.init.scripts.exists "nfs" true
Execute	.target.bash "/etc/init.d/nfs status" $["TERM":"raw"] 0
Read	.init.scripts.exists "SuSEfirewall2_init" true
Execute	.target.bash "/etc/init.d/SuSEfirewall2_init status" $["TERM":"raw"] 0
Return	true
Dump	
Dump	Read  - nfs not used & running
Dump	
Read	.etc.fstab [$["file":"/", "freq":1, "mntops":"defaults", "passno":2, "spec":"/dev/hda6", "vfstype":"ext3"]]
Read	.sysconfig.nfs.NFS4_SUPPORT "no"
Read	.sysconfig.nfs.NFS_SECURITY_GSS "no"
Read	.etc.idmapd_conf "Domain" "localhost"
Return	true
Dump	
Dump	Write - nfs not used - so, stopping services
Dump	
Read	.etc.fstab [$["file":"/", "freq":1, "mntops":"defaults", "passno":2, "spec":"/dev/hda6", "vfstype":"ext3"]]
Execute	.target.bash "/bin/cp $ORIG $BACKUP" $["BACKUP":"/etc/fstab.YaST2.save", "ORIG":"/etc/fstab"] 0
Write	.etc.fstab [$["file":"/", "freq":1, "mntops":"defaults", "passno":2, "spec":"/dev/hda6", "vfstype":"ext3"]] true
Write	.sysconfig.nfs.NFS4_SUPPORT "no" true
Write	.sysconfig.nfs.NFS_SECURITY_GSS "no" true
Read	.init.scripts.exists "nfs" true
Execute	.target.bash_output "/etc/init.d/nfs stop" $["TERM":"raw"] $["exit":0, "stderr":"", "stdout":""]
Read	.init.scripts.exists "SuSEfirewall2_init" true
Execute	.target.bash "/etc/init.d/SuSEfirewall2_init status" $["TERM":"raw"] 0
Return	true
Dump	
Dump	Read  - nfs not used & services are stopped
Dump	
Read	.etc.fstab [$["file":"/", "freq":1, "mntops":"defaults", "passno":2, "spec":"/dev/hda6", "vfstype":"ext3"]]
Read	.sysconfig.nfs.NFS4_SUPPORT "no"
Read	.sysconfig.nfs.NFS_SECURITY_GSS "no"
Read	.etc.idmapd_conf "Domain" "localhost"
Return	true
Dump	
Dump	Write - nfs not used; leave services stopped
Dump	
Read	.etc.fstab [$["file":"/", "freq":1, "mntops":"defaults", "passno":2, "spec":"/dev/hda6", "vfstype":"ext3"]]
Execute	.target.bash "/bin/cp $ORIG $BACKUP" $["BACKUP":"/etc/fstab.YaST2.save", "ORIG":"/etc/fstab"] 0
Write	.etc.fstab [$["file":"/", "freq":1, "mntops":"defaults", "passno":2, "spec":"/dev/hda6", "vfstype":"ext3"]] true
Write	.sysconfig.nfs.NFS4_SUPPORT "no" true
Write	.sysconfig.nfs.NFS_SECURITY_GSS "no" true
Read	.init.scripts.exists "nfs" true
Execute	.target.bash_output "/etc/init.d/nfs stop" $["TERM":"raw"] $["exit":0, "stderr":"", "stdout":""]
Read	.init.scripts.exists "SuSEfirewall2_init" true
Execute	.target.bash "/etc/init.d/SuSEfirewall2_init status" $["TERM":"raw"] 0
Return	true
